File: app/src/main/java/org/michaelbel/moviemade/data/eventbus/RxBus.java
package org.michaelbel.moviemade.data.eventbus;

import io.reactivex.Observable;
import io.reactivex.subjects.PublishSubject;

public class RxBus {

    public RxBus() {}

    private PublishSubject<Object> publishSubject = PublishSubject.create();

    public void send(Object object) {
        publishSubject.onNext(object);
    }

    public Observable<Object> toObservable() {
        return publishSubject;
    }
}

File: app/src/main/java/org/michaelbel/moviemade/data/eventbus/RxBus.kt
package org.michaelbel.moviemade.data.eventbus

import io.reactivex.Observable
import io.reactivex.subjects.PublishSubject

class RxBus {

    private val publishSubject = PublishSubject.create<Any>()

    fun send(`object`: Any) {
        publishSubject.onNext(`object`)
    }

    fun toObservable(): Observable<Any> {
        return publishSubject
    }
}